### PR TITLE
fix(prow/config): disable needs-rebase plugin

### LINKED ISSUE #961: need-rebase plugin will not remove label when conflicts solved
## Bug Report

need rebase can not remove the `needs-rebase` label automatically when the pull request has no conflicts

### PR DESCRIPTION
**Pull request by wuhuizuo**
Signed-off-by: wuhuizuo <wuhuizuo@126.com>

### What problem does this PR solve?

Close #961 

Previously, we tried to upgrade the `needs-rebase` plugin to the latest upstream version but still couldn't solve the problem. The analysis show it was unnecessary to enable in on repositories:

- Tide will determine whether there is a conflict with this PR when merging. (This is checked using the GitHub API.)
- There is also interception on the GitHub side, and conflicted pull requests cannot be merged.
- Of course, the current function of this label is to facilitate filtering which PR needs to handle conflicts. However, it seems to be misleading at present, and the value becomes negative. Also we can get the conflicted PRs simply with `gh` CLI tool:
  ```bash
  gh pr list --repo <OWNER>/<REPO> --state open --json number,title,mergeable --limit 1000 --jq '.[] | select(.mergeable == "CONFLICTING")'
  ```

In summary, it can be offline.

### What is changed and how it works?

What's Changed:
- remove the plugin for repos.

How it Works:

### CODE REVIEW

**Review by rleungx — APPROVED** [review]

**Review by BornChanger — APPROVED** [review]

**Review by purelind — APPROVED** [review]
/lgtm
